Add get-deps script

olexii4 opened the pull request:
Signed-off-by: Oleksii Orel <oorel@redhat.com>
Add get-deps script.

olexii4 committed:
add get-deps script

Signed-off-by: Oleksii Orel <oorel@redhat.com>

akurinnoy reviewed (changes requested):
@olexii4 

I see that `get-deps.js` completely rewrites `dev.md` and `prod.md` on each run. But we've discussed to only update lines with new or upgraded dependencies.

olexii4 commented:
@akurinnoy @sleshchenko I have rebased and added several real resolved CQs as an example. Just have a try from the root:
```sh
$ node .deps/get-deps.js
```
WDYT?

sleshchenko commented on .deps/prod.md:
```diff
@@ -0,0 +1,230 @@
+| `@babel/runtime@7.10.2` |  |
+| `@babel/types@7.10.2` |  |
+| `@eclipse-che/api@7.4.0` |  |
+| `@eclipse-che/workspace-client@0.0.1-1585913592` |  |
```
CQ is not needed here because it's eclipse-che part, right?

akurinnoy replied:
AFAIK, yes

olexii4 replied:
Right. It is  https://github.com/eclipse/che-workspace-client.
This information can be added as a comment instead of a link for CQ.

akurinnoy commented on .deps/dev.md:
```diff
@@ -0,0 +1,847 @@
+| `@babel/traverse` |  |
+| `@bcoe/v8-coverage` |  |
+| `@cnakazawa/watch` |  |
+| `@eclipse-che/api@7.5.0-SNAPSHOT` |  |
```
regex that removes versions needs to be improved

akurinnoy commented on .deps/get-deps.js:
```diff
@@ -0,0 +1,70 @@
+  // table body
+  depsArray.forEach(item => {
+    if (isDev) {
+      item = item.replace(/@[0-9]+.[0-9]+.[-0-9]+$/, '');
```
Try to use `/@[^@]+$/` instead

olexii4 replied:
ok

olexii4 replied:
I came back to save the version.

akurinnoy commented on .deps/get-deps.js:
```diff
@@ -0,0 +1,70 @@
+      // add a CQ
+      document += depsMap.get(item);
+    }
+    document += ' |\n';
```
I'd like you to use a template string here, i.e. 
```javascript
document += `| \`${item}\` | ${cq} |\n`
```

olexii4 replied:
done

sleshchenko commented on .deps/prod.md:
```diff
@@ -0,0 +1,230 @@
+| `color-name@1.1.3` |  |
+| `connected-react-router@6.8.0` |  |
+| `convert-source-map@1.7.0` |  |
+| `core-js@2.6.11` |  |
```
```suggestion
| `core-js@2.6.11` | https://dev.eclipse.org/ipzilla/show_bug.cgi?id=19151 |
```

olexii4 replied:
done

olexii4 committed:
fixup! add get-deps script

olexii4 committed:
fixup! add get-deps script
